zulip/integrations/matrix/matrix_bridge.py

```diff
@@ -77,30 +79,64 @@ def _matrix_to_zulip(room: Any, event: Dict[str, Any]) -> None:
 
     return _matrix_to_zulip
 
-def get_message_content_from_event(event: Dict[str, Any]) -> str:
+def get_message_content_from_event(event: Dict[str, Any]) -> Optional[str]:
+    irc_nick = shorten_irc_nick(event['sender'])
     if event['type'] == "m.room.member":
+        return None
+        # uncomment the lines below and delete the above `return`
+        # to display joining and leaving messages
+        """
         if event['membership'] == "join":
-            content = "{0} joined".format(event['sender'])
+            content = "**{0}** joined".format(irc_nick)
         elif event['membership'] == "leave":
-            content = "{0} quit".format(event['sender'])
+            content = "**{0}** quit".format(irc_nick)
+        """
     elif event['type'] == "m.room.message":
         if event['content']['msgtype'] == "m.text" or event['content']['msgtype'] == "m.emote":
-            content = "{0}: {1}".format(event['sender'], event['content']['body'])
+            content = "**{0}**: {1}".format(irc_nick, event['content']['body'])
     else:
         content = event['type']
     return content
 
+def shorten_irc_nick(nick: str) -> str:
+    """
+    Add nick shortner functions for specific IRC networks
+    Eg: For freenode change '@freenode_user:matrix.org' to 'user'
+    Check the list of IRC networks here:
+    https://github.com/matrix-org/matrix-appservice-irc/wiki/Bridged-IRC-networks
+    """
+    # for freenode
```

Comment:
First impression: could this be simplified by using a `dict` or `list` of regexes to work through?

This information is being "lost" on presentation, but presumably the config file is specifying this data anyhow, so:
* can we use the config info to do the regex, without hardcoding this?
* can the bot info as presented on zulip, absorb this information somehow, to make it clear which network(server)/channel the bot is connecting to?

Reply:
I investigated some more, and the networks [mentioned](https://github.com/matrix-org/matrix-appservice-irc/wiki/Bridged-IRC-networks) fit into a general pattern, and I have added the code for that. see: https://github.com/zulip/python-zulip-api/pull/396/commits/72175036c1ce4eb3ab1fcf356144a92f99ffd86d